lib/sparesults/src/error.rs

```diff
@@ -73,28 +40,26 @@ impl From<quick_xml::Error> for ParseError {
             quick_xml::Error::Io(error) => {
                 Self::Io(Arc::try_unwrap(error).unwrap_or_else(|e| io::Error::new(e.kind(), e)))
             }
-            _ => Self::Syntax(SyntaxError {
-                inner: SyntaxErrorKind::Xml(error),
-            }),
+            _ => Self::Syntax(SyntaxError::Xml(error)),
         }
     }
 }
 
 /// An error in the syntax of the parsed file.
-#[derive(Debug)]
-pub struct SyntaxError {
-    pub(crate) inner: SyntaxErrorKind,
-}
-
-#[derive(Debug)]
-pub(crate) enum SyntaxErrorKind {
-    Json(json_event_parser::SyntaxError),
-    Xml(quick_xml::Error),
+#[derive(Debug, thiserror::Error)]
+pub enum SyntaxError {
```

Comment:
I would prefer to make variants private here for encapsulation. They are mostly implementation details and not very meaningful for the user